pynamodb/models.py

```diff
@@ -946,43 +935,57 @@ def _get_save_args(self, attributes=True, null_check=True):
         if range_key_attribute:
             range_key = attribute_values.pop(range_key_attribute.attr_name, {}).get(range_key_attribute.attr_type)
         args = (hash_key, )
+        kwargs = {}
         if range_key is not None:
             kwargs['range_key'] = range_key
-        if attributes:
-            kwargs['attributes'] = attribute_values
+        version_condition = self._handle_version_attribute(attributes=attribute_values)
+        if version_condition is not None:
+            condition &= version_condition
+        kwargs['attributes'] = attribute_values
+        kwargs['condition'] = condition
         return args, kwargs
 
-    def _handle_version_attribute(self, serialized_attributes, actions=None):
+    def _get_hash_range_key_serialized_values(self) -> Tuple[Any, Optional[Any]]:
+        if self._hash_keyname is None:
+            raise Exception("The model has no hash key")
```

Comment:
Seems like precedent is a `ValueError` - https://github.com/pynamodb/PynamoDB/blob/6d6a8be7a25abf6f0ab5df7d8697c9fe871f7c4c/pynamodb/models.py#L304

Reply:
Considered that, but that actually relates to an argument that was passed in.

also lol `{range_key}`